core/domain/services/admin/entities/DefaultPrices.php

```diff
@@ -0,0 +1,109 @@
+use ReflectionException;
+use RuntimeException;
+
+/**
+ * Class DefaultPrices
+ * Description
+ *
+ * @package EventEspresso\core\domain\services\admin\events\editor
+ * @author  Brent Christensen
+ * @since   $VID:$
+ */
+class DefaultPrices implements DefaultEntityGeneratorInterface
+{
+
+    /**
+     * @var EEM_Price $price_model
+     */
+    protected $price_model;
+
+
+    /**
+     * @param EEM_Price      $price_model
+     */
+    public function __construct(EEM_Price $price_model)
+    {
+        $this->price_model = $price_model;
+    }
+
+
+    /**
+     * @param EE_Ticket|EE_Base_Class $entity
+     * @return EE_Price[]
+     * @throws EE_Error
+     * @throws InvalidInterfaceException
+     * @throws ReflectionException
+     * @since $VID:$
+     */
+    public function create(EE_Base_Class $entity)
+    {
+        if (! $entity instanceof EE_Ticket) {
+            throw new InvalidEntityException($entity, 'EE_Ticket');
+        }
+        $new_prices = [];
+        $has_base_price = false;
+        $default_prices = $this->price_model->get_all_default_prices();
+        if (is_array($default_prices)) {
+            foreach ($default_prices as $default_price) {
+                if (! $default_price instanceof EE_Price) {
+                    throw new InvalidEntityException($default_price, 'EE_Price');
+                }
+                $default_price_clone = clone $default_price;
+                $default_price_clone->set('PRC_ID', null);
+                $default_price_clone->set('PRC_is_default', false);
+                $default_price_clone->save();
+                $default_price_clone->_add_relation_to($entity, 'Ticket');
+                // verify that a base price has been set
+                $has_base_price = $default_price_clone->is_base_price() ? true : $has_base_price;
+                $new_prices[ $default_price_clone->ID() ] = $default_price_clone;
+            }
+        }
+        if (! $has_base_price) {
+            $new_base_price = $this->createNewBasePrice($entity);
+            $new_prices[ $new_base_price->ID() ] = $new_base_price;
+        }
+        return $new_prices;
+    }
+
+
+    /**
+     * @param EE_Ticket $ticket
+     * @return EE_Price
+     * @throws EE_Error
+     * @throws ReflectionException
+     * @since $VID:$
+     */
+    protected function createNewBasePrice(EE_Ticket $ticket)
+    {
+        $new_base_price = $this->price_model->get_new_price();
+        $base_price_type = $this->price_model->get_one([
+            'PRT_ID' => EEM_Price_Type::base_type_base_price
+        ]);
+        if (! $base_price_type instanceof EE_Price_Type) {
+            throw new RuntimeException(
+                esc_html__(
+                    'A valid base price type could not be retrieved from the database.',
+                    'event_espresso'
+                )
+            );
+        }
+        $new_base_price->set('PRT_ID', $base_price_type->ID());
```

Comment:
```suggestion
        $new_base_price->set('PRT_ID', EEM_Price_Type::base_type_base_price);
```

Reply:
no this is not correct

`PRT_ID` is the database ID for the actual price type record, whereas `EEM_Price_Type::base_type_base_price` corresponds to the `PBT_ID` 


PRT_ID | PRT_name | PBT_ID
-- | -- | --
1 | Base Price | 1
2 | Percent Discount | 2
3 | Dollar Discount | 2
4 | Percent Surcharge | 3
5 | Dollar Surcharge | 3
6 | Regional Tax | 4
7 | Federal Tax | 4

so although by default... on a fresh install of EE, `PRT_ID` === `PBT_ID` (`EEM_Price_Type::base_type_base_price`) 

if someone deletes those records and/or mucks with this table in their db, then `PRT_ID` for the base price type can end up being something completely different. This is why I was actually pulling the base price type record from the db and then using its actual `PRT_ID`.

We've had this issue happen in the past where a user had manually deleted these records from the db, which were then repopulated on the next EE request but with new higher auto-incremented IDs. Needless to say, things went 💥 !

Reply:
Thanks 👍

Reply:
Then we first need to get the pricetype object and use its ID